AUTHORING_GUIDE.md

```diff
@@ -1,207 +1,117 @@
-# distributed under the License is distributed on an "AS IS" BASIS,
-# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-# See the License for the specific language governing permissions and
-# limitations under the License.
-```
-
-### Module-level docstring
-
-All samples should contain a module-level docstring. For command-line
-applications, this docstring will be used as the summary when `-h` is passed.
-The docstring should be succinct and avoid repeating information
-available in readmes or documentation.
-
-Here's a simple docstring for a command-line application with straightforward
-usage:
-
-```
-This application demonstrates how to perform basic operations on blobs
-(objects) in a Google Cloud Storage bucket.
-
-For more information, see the README.md under /storage and the documentation
-at https://cloud.google.com/storage/docs.
-```
-
-Here's a docstring from a command-line application that requires a little
-bit more explanation:
-
-```python
-"""This application demonstrates how to upload and download encrypted blobs
-(objects) in Google Cloud Storage.
+If you prefer to use [pylint](https://www.pylint.org/), note that Python
+samples for this repo are not required to conform to pylint’s default settings
+outside the scope of PEP 8, such as the “too many arguments” or “too many
+local variables” warnings.
 
-Use `generate-encryption-key` to generate an example key:
+The use of [Black](https://pypi.org/project/black/) to standardize code
+formatting and simplify diffs is recommended, but optional.
 
-    python encryption.py generate-encryption-key
+In addition to the syntax guidelines covered in PEP 8, samples should strive
+to follow the Pythonic philosophy outlined in the
+[PEP 20 - Zen of Python](https://www.python.org/dev/peps/pep-0020/) as well
+as the readability tenets presented in Donald Knuth's
+_[Literate Programming](https://en.wikipedia.org/wiki/Literate_programming)_.
+Notably, your sample program should be self-contained, readable from top to
+bottom, and fairly self-documenting. Prefer descriptive names, and use
+comments and docstrings only as needed to further clarify the code’s intent.
+Always introduce functions and variables before they are used. Prefer less
+indirection. Prefer imperative programming as it is easier to understand.
 
-Then use the key to upload and download files encrypted with a custom key.
-
-For more information, see the README.md under /storage and the documentation
-at https://cloud.google.com/storage/docs/encryption.
-"""
-```
-
-Finally, here's a docstring from a web application:
-
-```python
-"""Google Cloud Endpoints sample application.
-
-Demonstrates how to create a simple echo API as well as how to use the
-various authentication methods available.
-
-For more information, see the README.md under /appengine/flexible and the
-documentation at https://cloud.google.com/appengine/docs/flexible.
-"""
-```
-
-### Functions & classes
+## Functions and Classes
```

Comment:
This section seems to contain several things that might be better to break up into smaller sections so its easier to link/reference:
* Prefer descriptive function names
* rules around documenting arguements
* Mentions types, but I think we should embrace the py3 type annotations `def hello_world(name: string):`